CMakeLists.txt

```diff
@@ -79,6 +79,40 @@ ELSE()
   OPTION(WITH_UPDATER "Include libraries and code to allow release versions to be updated on demand" ON)
 ENDIF()
 
+# Enable the inclusion of fonts currently carried in the source code by default
+# unless the environmental variable WITH_FONTS is already defined and is not
+# set to (case insenstive) "no". Linux packagers will find it useful to do this
+# since package managers may already package the relevant fonts - or are not
+# willing or able to include them.
+# Note: WITH_FONTS is an environmental value/variable (could be a number, a
+#           string, something else or not even exisit).
+#       FONT_TEST and USE_FONTS are CMake variables (probably a string and a
+#           boolean {ON or OFF} respectively).
+#       INCLUDE_FONTS is a C preprocessor symbol (i.e. the same as
+#           "#define INCLUDE_FONTS" which, in the absence of an explicitly set
+#           value, takes the numeric value of (int)1 as far as the compiler is
+#           concerned!)
```

Comment:
Could we condense this comment to just `to disable font inclusion in the binary, set the environment variable INCLUDE_FONTS to "no"`

Reply:
I wanted to explain the differences between the entities I was using - there are differences between how they behave (AFAICT) and if something doesn't work in the future (or if someone is porting to a different platform) it does not hurt to explain what is supposed to be going on; and, although the intention was to have the same action in both the cmake and the qmake cases there are slight differences - again AFAICT.

OTOH I suppose that, as the bottom line is that if the user does not do anything the default behaviour is as before, those who will want to tweak the settings are more likely to have a handle on the knobs to frob. 🤔

Reply:
I still want to keep it simple. Having lengthy explanations hurts because people read them even if they don't need to, get confused by them, become unsure if they were supposed to understand them, and then think the whole project is complicated.

Reply:
Conversely though, having undocumented options that are not explained is even worse in my book having properly commented stuff (and meaningful variable names) is always better than obscure settings whose effects can make things unworkable.  It is a fine line though, and one that is subject to a certain amount of personal opinion.

Reply:
I did not propose to remove the text entirely - see my original comment:

> Could we condense this comment to just `to disable font inclusion in the binary, set the environment variable INCLUDE_FONTS to "no"`

Even simpler - and more correct, yes I also got confused by the explanation that was explaining far too much, just this would be super clear:

> to disable font embedding in the binary, do: export WITH_FONTS="no"

Why not keep things super simple and crystal clear?

Think about your audience. Whom are you writing this for? The Linux packager, because that's the only person who has a practical use for this variable. When their goal is to disable the built-in fonts to comply with the licensing guidelines, do they need to know or worry about CMake variables or the C preprocessor variables? Not really. All they want to do is disable the fonts and they want to learn how to do that in the fastest way possible. Hence my suggestion that goes straight to the point!

Let me know what you think :smile:

Reply:
OK, will rethink this...

Reply:
... have done something here!